broadinstitute/scp-ingest-pipeline
# Fix gene names with slashes in expression writer (SCP-5341)

This enables the expression data caching pipeline to handle gene names with slashes and other non-URL-safe characters.

Previously, gene names like "THRA1/BTR" were written to the bucket in a raw manner, which caused downstream problems when trying to fetch JSON data for those genes.  Slashes are not part of standard gene names, but they are in some gene name _aliases_, which have been observed in production.  This prevented those caches from being retrieved in the client, and broke Image Pipeline.

Now, slashes and other such characters are safely URL-encoded, letting downstream code fetch e.g. "THRA1%2FBTR.json".

### Test
Automated tests have been enhanced to confirm this handling.  If you want to manually test:
* Complete test instructions per PR #273 (but note `setup_mongo_dev.sh` is now `setup-mongo-dev.sh`)
* In your browser, go to your Google bucket
* Navigate to `_scp_internal/cache/expression_scatter/data/Dense_Example`
* Confirm `THRA1%2FBTR.json` exists there

This relates to SCP-5341.
